lib/cpp/export/ecl/ostream.hpp

```diff
@@ -2,190 +2,170 @@
+ {
+     
+    char buff[32];
+    
+    int n = snprintf(buff, sizeof(buff), "%u", value);
+    
+    ecl_assert(n > 0 && n < static_cast<int>(sizeof(buff)));
+
+    m_device->write(reinterpret_cast<uint8_t*>(buff), n);
+        
     return *this;
-}
+ }
+ 
+ 
+ template< class IO_device >
+ ostream<IO_device> &ostream< IO_device >::operator<<(char character)
+ {
+     
+    char buff [32];
+    
+    char n = snprintf(buff, sizeof(buff), "%c", character);
 
+    ecl_assert(n > 0 && n < char(sizeof(buff)));
 
-template< class IO_device >
-ostream<IO_device> &ostream< IO_device >::operator<< (unsigned int value)
-{
-    unsigned int higher_multiplicand = 1;
-    unsigned int out_digit = 0;
-    unsigned int val2 = value;
-    char out_character;
-
-    while (val2 / 10 != 0) {
-        higher_multiplicand *= 10;
-        val2 /= 10;
-    }
-
-    int delim = higher_multiplicand;
-
-    for (unsigned int i = 1; i <= higher_multiplicand; i *= 10) {
-        out_digit = value / delim;
-        out_character = out_digit + 48;
-
-        if (m_device->write((uint8_t *) &out_character, 1) < 0) {
-            break; //FIXME: add error handling
-        }
-
-        value = value - out_digit * delim;
-        delim /= 10;
-    }
+    m_device->write((uint8_t*)(buff), n);
 
     return *this;
-}
-
-
-template< class IO_device >
-ostream<IO_device> &ostream< IO_device >::operator<<(char character)
-{
-    if (m_device->write((uint8_t *) &character, 1) < 0) {
-        return *this; //FIXME: add error handling
-    }
-
-    return *this;
-}
-
-
-template< class IO_device >
-ostream<IO_device> &ostream< IO_device >::operator<<(const char *string)
-{
-    // TODO: improve this function in a way it can write a string
-    // to a device with chunks that are split by '\n' symbol
-    // What about somehow move it to the platform layer?
-
-    size_t i = 0;
-    while (string[i] != '\0') {
-        if (string[i] == '\n') {
-            uint8_t carret_ret = '\r';
-            m_device->write(&carret_ret, 1);
-        }
-
-        if (m_device->write((const uint8_t *) &string[i], 1) < 0) {
-            //Serial.printlnf("\n>%d<\n", count);
-            break; //FIXME: add error handling
-        }
-
-        i++;
-    }
-    return *this;
-}
-
-
-template< class IO_device >
-ostream<IO_device> &ostream< IO_device >::operator<<(
-        ostream& (*func)(ostream< IO_device >&))
-{
-    return func(*this);
-}
-
-template< class IO_device >
-ostream<IO_device> &ostream< IO_device >::put(char c)
-{
-    m_device->write((uint8_t *)&c, 1);
-    return *this;
-}
-
-}
-
-#endif // ECL_OSTREAM_HPP
+ }
+ 
+ 
+ template< class IO_device >
+ ostream<IO_device> &ostream< IO_device >::operator<<(const char *string)
+ {
+     // TODO: improve this function in a way it can write a string
+     // to a device with chunks that are split by '\n' symbol
+     // What about somehow move it to the platform layer?
+ 
+     size_t i = 0;
+     while (string[i] != '\0') 
```

Comment:
According to the code style - place brackets on 1st line of `while` and similar statements.
See Rule 1. in https://forgge.github.io/theCore/community.html#bracket-placement